From c730317528731efb3b8977b6d293d20069e5b51b Mon Sep 17 00:00:00 2001
Message-Id: <c730317528731efb3b8977b6d293d20069e5b51b.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Nabendu Maiti <nabendu.bikash.maiti@intel.com>
Date: Wed, 24 Sep 2014 00:12:47 +0530
Subject: [PATCH 306/312] FOR_UPSTREAM [VPG]: drm/i915: Reduce delay in dsi
 enable time.

This patch reduces the time required to enable dsi in cold boot and
resume case in dsi_enble before sending dsi command.

Issue: GMIN-1464
Change-Id: Ie21eb6cfc54633327f2411b39a0d7d01f4c4ebb4
Signed-off-by: Nabendu Maiti <nabendu.bikash.maiti@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index b40e7e8..c017293 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -137,9 +137,8 @@ static void intel_dsi_enable(struct intel_encoder *encoder)
 	if (is_cmd_mode(intel_dsi))
 		I915_WRITE(MIPI_MAX_RETURN_PKT_SIZE(pipe), 8 * 4);
 	else {
-		msleep(20); /* XXX */
 		dpi_send_cmd(intel_dsi, TURN_ON, DPI_LP_MODE_EN);
-		msleep(100);
+		usleep_range(1500, 2000);
 
 		if (intel_dsi->dev.dev_ops->enable)
 			intel_dsi->dev.dev_ops->enable(&intel_dsi->dev);
-- 
1.7.9.5

